superset/commands/dashboard/importers/v0.py

```diff
@@ -252,20 +255,30 @@ def alter_native_filters(dashboard: Dashboard) -> None:
 
     alter_native_filters(dashboard_to_import)
 
-    new_slices = (
-        session.query(Slice).filter(Slice.id.in_(old_to_new_slc_id_dict.values())).all()
-    )
-
     if existing_dashboard:
         existing_dashboard.override(dashboard_to_import)
-        existing_dashboard.slices = new_slices
-        session.flush()
-        return existing_dashboard.id
+    else:
+        db.session.add(dashboard_to_import)
+
+    dashboard = existing_dashboard or dashboard_to_import
+    dashboard.slices = (
+        db.session.query(Slice)
+        .filter(Slice.id.in_(old_to_new_slc_id_dict.values()))
+        .all()
+    )
 
-    dashboard_to_import.slices = new_slices
-    session.add(dashboard_to_import)
+    # Migrate any filter-box charts to native dashboard filters.
+    migrate_dashboard(dashboard)
     session.flush()
```

Comment:
In actuality only the new (non-existing) dashboard strictly needs to be flushed (after being added to the Flask-SQLAlchemy session) to the database in order to have the dashboard ID assigned. I'm merely preserving the previous behavior by always flushing prior to returning.